### PR TITLE
Ramone/tools 1912

### PR DESCRIPTION
**Pull request by Ramone-HL**
Added the section, Build a dApp locally in the Build chapter

### CODE REVIEW

File: docs/build.md [hunk 1]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+## Build a dApp locally
+
+*Learn how to set up a local environment or a local Yuma testnet node to build a dApp (decentralized application).*
+
+In this section, you will accomplish the following:
+
+* Deploy an existing dApp into a local environment or Yuma Testnet
+* Run a private instance of EON 
+
+The easiest way to deploy a dApp is to use **Docker**. The EON client will be running in a Dockerized container. See **Node Client** below.
```
Comment (yurikoinaba) [inline docs/build.md:432]: Can we link `**Node Client**` to the section below : `/docs/build#node-client` ?

File: docs/build.md [hunk 2]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+
+Horizen Labs provides two separate projects to enable:
+
+* **SANDBOX:** creating a personal sandboxed blockchain 
+* **TESTNET:** starting your EON node
+
+For using a **personal sandboxed blockchain**, there is no required configuration. A local and disposable sidechain will be created from scratch with the latest available version of EON. Check the **.env** file for the port you will use for RPC calls. 
+
+Horizen Labs repositories are open source, including EON. The code can be pulled from the GitHub repository and executed, however this is not considered in the scope of the testnet documentation.
```
Comment (yurikoinaba) [inline docs/build.md:441]:
```suggestion
Horizen Labs repositories are open source, including EON. The code can be pulled from the GitHub repository and executed. However, this is not considered in the scope of the testnet documentation. 
```

File: docs/build.md [hunk 3]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+Horizen Labs provides two separate projects to enable:
+
+* **SANDBOX:** creating a personal sandboxed blockchain 
+* **TESTNET:** starting your EON node
+
+For using a **personal sandboxed blockchain**, there is no required configuration. A local and disposable sidechain will be created from scratch with the latest available version of EON. Check the **.env** file for the port you will use for RPC calls. 
+
+Horizen Labs repositories are open source, including EON. The code can be pulled from the GitHub repository and executed, however this is not considered in the scope of the testnet documentation. 
+
+**Note:** EON is considered a sample application of what can be built using our Sidechains-SDK, that’s why you will find it in the same repository.
```
Comment (yurikoinaba) [inline docs/build.md:443]:
```suggestion
**Note:** EON is considered a sample application of what can be built using our Sidechains-SDK, which is why you will find it in the same repository.
```

File: docs/build.md [hunk 4]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+
+At the time of this writing, EON is not finalized. It is recommended to use the **M3-V2** tag.  
+
+### Node Client
+To start your node, run our Docker compose (a tool required to run our multi-container application) on your computer. Next, open a terminal and send some simple commands.
```
Comment (yurikoinaba) [inline docs/build.md:448]:
```suggestion
To start your node, run our Docker compose (a tool required to run our multi-container application) on your computer. Next, open a terminal and execute some simple commands.
```

File: docs/build.md [hunk 5]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+
+#### Set Up
+
+1. Remove any previous version of Docker including any Docker-compose plugin before starting.
```
Comment (yurikoinaba) [inline docs/build.md:452]:
```suggestion
1. Remove any previous version of Docker including any Docker-compose plugins before starting.
```

File: docs/build.md [hunk 6]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+groupadd -g 999 docker 2>&1 || groupmod -g 999 docker
+```
+
+Get **'docker compose'** bash completion.
+
+```
+curl -fsSL https://raw.githubusercontent.com/docker/cli/master/contrib/completion/bash/docker -o /etc/bash_completion.d/docker
+```
+
+**Note:** Step 2 refers to Ubuntu or almost any Debian-based Linux distro. Check the official documentation for other OS or other kinds of incompatibilities.
```
Comment (yurikoinaba) [inline docs/build.md:489]: Can we link "official documentation"?

File: docs/build.md [hunk 7]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+
+4. Build the project by using the following command:
+
+```
+./scripts/init.sh
+```
+
+You are now set, the **client** starts automatically.
+
+**Note:** YUMA TESTNET requires time to synchronize the entire blockchain. While waiting, check the progress by comparing the last block in the **Yuma Explorer** with the response of the following rpc method:
```
Comment (yurikoinaba) [inline docs/build.md:512]:
```suggestion
**Note:** YUMA TESTNET requires time to synchronize the entire blockchain. While waiting, check the progress by comparing the last block in the **Yuma Explorer** with the response of the following RPC method:
```

File: docs/build.md [hunk 8]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+
+```
+curl -X POST "http://127.0.0.1:9545/ethv1/" --data '{"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":"test"}' -H "accept: application/json"
+```
+
+**WARNING:** **Risk Data Corruption** - Always shutdown the node properly by executing *shutdown.sh* available in the scripts folder at the end of your testing session and/or before turning off your machine (computer or virtual machine).
```
Comment (yurikoinaba) [inline docs/build.md:518]:
```suggestion
**WARNING:** **Risk Data Corruption** - Always shutdown the node properly by executing *shutdown.sh*, available in the scripts folder, at the end of your testing session and/or before turning off your machine (computer or virtual machine). 
```

File: docs/build.md [hunk 9]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+**WARNING:** **Risk Data Corruption** - Always shutdown the node properly by executing *shutdown.sh* available in the scripts folder at the end of your testing session and/or before turning off your machine (computer or virtual machine). 
+
+### Run the client
+If you have performed the previous setup instructions and you want to restart the client, run the following command at the root folder of the project (inside the cloned repository):
+
+```
+docker compose up -d
+```
+
+**WARNING:** **Risk Data Corruption** - Always shutdown the node properly by executing *shutdown.sh* available in the scripts folder at the end of your testing session and/or before turning off your machine (computer or virtual machine).
```
Comment (yurikoinaba) [inline docs/build.md:527]:
```suggestion
**WARNING:** **Risk Data Corruption** - Always shutdown the node properly by executing *shutdown.sh*, available in the scripts folder, at the end of your testing session and/or before turning off your machine (computer or virtual machine). 
```

File: docs/build.md [hunk 10]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+Set up your network provider as **localhost (127.0.0.1)** and **port (9545)**, and also change the **Chain ID** according to the environment you are using.
+
+##### YUMA SANDBOX
+
+```
+Network name:  Yuma Sandbox 
+New RPC URL:  http://127.0.0.1:9545/ethv1
+Chain ID: 1997 
+Currency symbol: ZEN
+Block Explorer: leave blank
```
Comment (yurikoinaba) [inline docs/build.md:539]:
```suggestion
Block Explorer: _leave blank_
```

File: docs/build.md [hunk 11]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+Network name:  Yuma Testnet Local
+New RPC URL:  http://127.0.0.1:9545/ethv1 
+Chain ID: 1662
+Currency symbol: ZEN
+Block Explorer: https://yuma-explorer.horizen.io/
+```
+
+**Note:** For configuring a wallet or network provider with the testnet local node the "RPC Server" must be the **local one (localhost)**. Use the **Chain ID** for  the  corresponding  testnet. 
+
+### Why Should I Run a node?
```
Comment (yurikoinaba) [inline docs/build.md:555]: It should either be `Why Should I Run a Node?` or `Why should I run a node?`

File: docs/build.md [hunk 12]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+
+Running your own node preserves your privacy and bolster your security. It also allows you to prove that no one is manipulating the transactions in EON or changing any rule.
```
Comment (yurikoinaba) [inline docs/build.md:557]:
```suggestion
Running your own node preserves your privacy and bolsters your security. It also allows you to prove that no one is manipulating the transactions in EON or changing any rules.
```

File: docs/build.md [hunk 13]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+
+**Note:** YUMA TESTNET requires time to synchronize the entire blockchain. While waiting, check the progress by comparing the last block in the **Yuma Explorer** with the response of the following rpc method:
+
+```
+curl -X POST "http://127.0.0.1:9545/ethv1/" --data '{"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":"test"}' -H "accept: application/json"
+```
+
+**WARNING:** **Risk Data Corruption** - Always shutdown the node properly by executing *shutdown.sh* available in the scripts folder at the end of your testing session and/or before turning off your machine (computer or virtual machine). 
+
+### Run the client
```
Comment (taijonsmith) [inline docs/build.md:520]: `client` should be capitalized

File: docs/build.md (hunk 8 above)
Comment (taijonsmith) [inline docs/build.md:518]: small, but `shutdown` should be `shut down`

File: docs/build.md [hunk 14]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+Horizen does not collect private information. However, by providing free access to the code, Horizen cannot guarantee that the same privacy with third-party operated dApps, and in either way the statement cannot be "**trustless**".
+
+### Security Benefits
+Using an EON node to create transactions can also improve your security by reducing or eliminating the need to expose your private keys to the internet through a web browser. The private key is required to sign the transactions before submission, while the signature generally takes place locally.
+
+**Note:** If you own a Validator node, you will benefit from the latest secure enclave features for key management, where the documentation of which will be shared privately. In this case, the signature takes place in a remote trusted execution environment.
```
Comment (taijonsmith) [inline docs/build.md:567]: `...key management, the documentation for which will be shared privately.`

File: docs/build.md [hunk 15]
```diff
@@ -420,3 +420,150 @@ npx hardhat compile
...
+**Note:** For configuring a wallet or network provider with the testnet local node the "RPC Server" must be the **local one (localhost)**. Use the **Chain ID** for  the  corresponding  testnet. 
+
+### Why Should I Run a node?
+
+Running your own node preserves your privacy and bolster your security. It also allows you to prove that no one is manipulating the transactions in EON or changing any rule.
+
+### Privacy Benefits
+
+When you run your own node, you can create and broadcast transactions directly from the node, thereby avoid using services that might compromise private information. A node also removes the need to use a block explorer to verify the status of your transactions. Block explorers allow third parties to track your transaction history by linking it to your IP address, leaking your physical location, your balance, and your financial counterparties.
+Horizen does not collect private information. However, by providing free access to the code, Horizen cannot guarantee that the same privacy with third-party operated dApps, and in either way the statement cannot be "**trustless**".
```
Comment (taijonsmith) [inline docs/build.md:562]: The last sentence is a bit confusing to read. Maybe remove `that`

File: docs/deploy.md [hunk 16]
```diff
@@ -209,8 +209,29 @@ truffle(zen)> withdrawalContract.submitWithdrawalRequest(mcAddress, { value: 100
 
 **Note:** The mcAddress needs to be defined or replaced by the string containing the Horizen mainchain address. 
 
-The target mainchain address of a backward transfer has to be a valid address: starting with “**zn**” for *mainnet* or “**zt**” for *testnet*.
+The target mainchain address of a backward transfer must be a **valid address** with the following format:
 
+1. Use **zn** for *mainnet* or **zt** for *testnet*.
+
+
+    **Example:** ztWBHD2Eo6uRLN6xAYxj8mhmSPbUYrvMPwt
```
Comment (taijonsmith) [inline docs/deploy.md:217]: I don't think the `Example` lines are being formatted correctly on the site https://ramone-tools-1912.evm-documentation.pages.dev/docs/deploy/#interact-with-native-smart-instance

Comment (Ramone-HL) [reply docs/deploy.md:217]: made the adjustment to the Example lines

**Review by taijonsmith — APPROVED** [review]

**Review by yurikoinaba — APPROVED** [review]

**Pull request merged**